webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js

```diff
@@ -49,27 +56,36 @@ class DatePicker extends React.Component {
       </Popover>
     );
     return (
-      <div>
+      <InputGroup className="input-group date-time-picker-pf">
+        <FormControl
+          aria-label="date-time-picker-input"
+          type="text"
+          className="date-input"
+          value={hiddenValue ? '' : this.formatDate()}
+          name={name}
+          onChange={e => this.setSelected(e.target.value)}
+        />
         <OverlayTrigger
           trigger="click"
           placement={placement}
           overlay={popover}
           rootClose
+          onEnter={() => this.setState({ hiddenValue: false })}
         >
-          <InputGroup className="input-group date-time-picker-pf">
-            <FormControl
-              aria-label="date-time-picker-input"
-              type="text"
-              value={this.formatDate()}
-              name={name}
-              onChange={e => this.setSelected(e.target.value)}
-            />
-            <InputGroup.Addon className="date-picker-pf">
-              <Icon type="fa" name="calendar" />
-            </InputGroup.Addon>
-          </InputGroup>
+          <InputGroup.Addon className="date-picker-pf">
+            <Icon type="fa" name="calendar" />
+          </InputGroup.Addon>
         </OverlayTrigger>
-      </div>
+        <InputGroup.Addon className="clear-button">
+          <Icon
+            type="fa"
+            name="close"
+            onClick={() =>
+              this.setState({ hiddenValue: true, value: new Date() })
```

Comment:
What if a date input is required ? this clear button becomes redundant?

Reply:
well, I guess we also need to improve it in `DateTimePicker.js`